shared/gateway/api_middleware.go

```diff
@@ -61,8 +61,9 @@ type CustomHandler = func(m *ApiProxyMiddleware, endpoint Endpoint, w http.Respo
 
 // HookCollection contains hooks that can be used to amend the default request/response cycle with custom logic for a specific endpoint.
 type HookCollection struct {
-	OnPreDeserializeRequestBodyIntoContainer  []Hook
-	OnPostDeserializeRequestBodyIntoContainer []Hook
+	OnPreDeserializeRequestBodyIntoContainer      []Hook
+	OnPostDeserializeRequestBodyIntoContainer     []Hook
+	OnPreDeserializeGrpcResponseBodyIntoContainer []func([]byte, interface{}) (bool, ErrorJson)
```

Comment:
Not super happy that the type is not `Hook`, but one generic func type just won't cut it everywhere. This might warrant a redesign, but I'd rather look into it after shipping Altair.